AdyenActions/Components/3DS2/Action handlers/3DS2 Without Delegated Authentication/ThreeDS2CoreActionHandler.swift

```diff
@@ -283,22 +296,20 @@ internal class ThreeDS2CoreActionHandler: AnyThreeDS2CoreActionHandler {
     
     // MARK: - Events {
     
-    private func sendFingerPrintEvent(_ subtype: AnalyticsEventLog.LogSubType) {
+    private func sendLogEvent(_ subtype: AnalyticsEventLog.LogSubType, for component: String) {
         let logEvent = AnalyticsEventLog(
-            component: Constants.fingerprintEvent,
+            component: component,
             type: .threeDS2,
             subType: subtype
         )
         context.analyticsProvider?.add(log: logEvent)
     }
     
-    private func sendChallengeEvent(_ subtype: AnalyticsEventLog.LogSubType) {
-        let logEvent = AnalyticsEventLog(
-            component: Constants.challengeEvent,
-            type: .threeDS2,
-            subType: subtype
-        )
-        context.analyticsProvider?.add(log: logEvent)
+    private func sendErrorEvent(_ code: AnalyticsConstants.ErrorCode, for component: String, message: String? = nil) {
```

Comment:
There is some inconsistency between method name and it's parameters.
First one with name omitted is code and from the outside it suggests and event (sendError**Event**).
I would suggest to make it more consistent to not accumulate ambiguity.

It also does two things at a time: 
- creates an instance of `AnalyticsEventError` which is not a direct responsibility of `ThreeDS2CoreActionHandler`
- asks to send it 

I see we now have `AnalyticsEventError` instantiation across multiple files, do you think we can extract it so in the end we only send events using a predefined enum(s) from specific handlers such as `ThreeDS2CoreActionHandler`?